File: Python/_110_Balanced_Binary_Tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def __init__(self):
        self.balanced = True
    
    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        self.maxDepth(root)
        return self.balanced
        
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        else:
            left_depth = self.maxDepth(root.left) + 1
            right_depth = self.maxDepth(root.right) + 1
            if abs(left_depth - right_depth) > 1:
                self.balanced = False
            return max(left_depth, right_depth)

File: Java/_110_Balanced_Binary_Tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    
    public boolean balanced = true;
    
    public boolean isBalanced(TreeNode root) {
        maxDepth(root);
        return balanced;
    }
    
    
    public int maxDepth(TreeNode root) {
        if (root == null)
            return 0;
        else
        {
            int leftDepth = maxDepth(root.left) + 1;
            int rightDepth = maxDepth(root.right) + 1;
            if (Math.abs(leftDepth - rightDepth) > 1)
                balanced = false;
            return Math.max(leftDepth, rightDepth);
        }
    }
}